src/core/providers/ogr/qgsogrprovider.cpp

```diff
@@ -1890,13 +1897,32 @@ bool QgsOgrProvider::addFeatures( QgsFeatureList &flist, Flags flags )
 
   const bool inTransaction = startTransaction();
 
+  QgsFeatureId incrementalFeatureId = -1;
+  OGRGeometryH filter;
```

Comment:
below block could be restricted to ``if ( !( flags & QgsFeatureSink::FastInsert ) )``